weeklyChallenges04/ChallengesSet04.cs

```diff
@@ -1,52 +1,132 @@
 ﻿using System;
+using System.Collections.Generic;
+using System.Linq;
 namespace weeklyChallenges04
 {
     public class ChallengesSet04
     {
         public int AddEvenSubtractOdd(int[] numbers)
         {
-            throw new NotImplementedException();
+            int total = 0;
+            foreach (var item in numbers)
+            {
+                if (item % 2 == 0)
+                {
+                    total += item;
+                }
+                else
+                {
+                    total -= item;
+                }
+            }
+            return total;
         }
 
         public int GetLengthOfShortestString(string str1, string str2, string str3, string str4)
         {
-            throw new NotImplementedException();
+            List<string> Thane = new List<string>();
+            Thane.Add(str1);
+            Thane.Add(str2);
+            Thane.Add(str3);
+            Thane.Add(str4);
+            string[] Neymor = Thane.ToArray() ;
+            int minLength = Neymor.Min(y => y.Length);
+            return minLength;
+            
         }
 
         public int GetSmallestNumber(int number1, int number2, int number3, int number4)
         {
-            throw new NotImplementedException();
+            List<int> Bobby = new List<int>(number1);
+            Bobby.Add(number1);
+           Bobby.Add(number2);
+           Bobby.Add(number3);
+           Bobby.Add(number4);
+            int[] Greg = Bobby.ToArray();
+            return Greg.Min(x => x);
         }
 
         public void ChangeBusinessNameTo_TrueCoders(Business biz)
         {
-            throw new NotImplementedException();
+           biz.Name = "TrueCoders";
         }
 
         public bool CouldFormTriangle(int sideLength1, int sideLength2, int sideLength3)
         {
-            throw new NotImplementedException();
+            if (sideLength1 + sideLength2 > sideLength3 && sideLength1+sideLength3 > sideLength2 && sideLength2+sideLength3>sideLength1)
```

Comment:
Since this is a bool, u could have simply written:

`return (sideLength1 + sideLength2 > sideLength3 && sideLength1+sideLength3 > sideLength2 && sideLength2+sideLength3>sideLength1)`